src/main/kotlin/no/nav/syfo/aktivitetskrav/database/AktivitetskravRepository.kt

```diff
@@ -0,0 +1,90 @@
+package no.nav.syfo.aktivitetskrav.database
+
+import no.nav.syfo.application.database.DatabaseInterface
+import no.nav.syfo.application.database.toList
+import no.nav.syfo.domain.PersonIdent
+import java.sql.ResultSet
+import java.time.OffsetDateTime
+import java.util.*
+
+class AktivitetskravRepository(private val database: DatabaseInterface) {
+
+    fun getAktivitetskrav(uuid: UUID): PAktivitetskrav? =
+        database.connection.prepareStatement(getAktivitetskravByUuidQuery).use {
+            it.setString(1, uuid.toString())
+            it.executeQuery().toList { toPAktivitetskrav() }.firstOrNull()
+        }?.run {
+            val vurderinger = getAktivitetskravVurderinger(aktivitetskravId = id)
+            copy(vurderinger = vurderinger)
+        }
+
+    fun getAktivitetskrav(
+        personIdent: PersonIdent,
+    ): List<PAktivitetskrav> =
+        database.connection.prepareStatement(getAktivitetskravByPersonidentQuery).use {
+            it.setString(1, personIdent.value)
+            it.executeQuery().toList { toPAktivitetskrav() }
+        }.map {
+            val vurderinger = getAktivitetskravVurderinger(aktivitetskravId = it.id)
+            it.copy(vurderinger = vurderinger)
+        }
+
+    private fun getAktivitetskravVurderinger(
+        aktivitetskravId: Int,
+    ): List<PAktivitetskravVurdering> =
+        database.connection.use { connection ->
```

Comment:
Trenger ikke endre det nå, men ser ut som vi kanskje kunne gjenbrukt connection fra funksjonen som henter aktivitetskravet her :thinking:

Reply:
Ja, siden dette er en private-metode synes jeg det hadde vært bedre å ha connection som en parameter sånn at den kan "gjenbrukes" fra metodene ovenfor.

Særlig for den siste: hvis en person har 5 aktivitetskrav vil jo dette kreve 6 separate transaksjoner, når man egentlig kan gjøre alt med en connection/transaksjon.

Reply:
Endrer til å gjenbruke connection 👍 Sånn optimalt sett skulle vi jo ha hentet alt i en query, men det fikk jeg ikke helt til🤔 Dere vet ikke hvordan man kan få til det her?

Reply:
Går nok an å join'e tabellene, men det er kanskje like greit å gjøre det sånn som det er nå - da blir query'ene enkle. 

Men kanskje det hadde vært bedre/penere hvis getAktivitetskravVurderinger var en extension-function på Connection?

Reply:
God idé! Så lenge denne er `private` så bør det ikke være noe problem. Endrer 🙏